lua/frecency/picker.lua

```diff
@@ -0,0 +1,264 @@
+---@return function
+---FIXME: path transacted with icons being pre-appended instead of appended
+M.maker = function(entry)
+  local items = {}
+  if M.config.show_scores then
+    table.insert(items, { width = 8 })
+  end
+
+  if M.config.show_filter_column then
+    local width = 0
+    -- TODO: Only add +1 if M.show_filter_thing is true, +1 is for the trailing slash
+    if M.active_filter and M.active_filter_tag == "LSP" then
+      width = #(utils.path_tail(M.active_filter)) + 1
+    elseif M.active_filter then
+      width = #(Path:new(M.active_filter):make_relative(os_home)) + 1
+    end
+    table.insert(items, { width = width })
+  end
+
+  if not M.config.disable_devicons and has_devicons then
+    table.insert(items, { width = 2 }) -- icon column
+  end
+
+  table.insert(items, { remaining = true })
+
+  return {
+    filename = entry.path,
+    ordinal = entry.path,
+    name = entry.path,
+    score = entry.score,
+    display = function(e)
+      return entry_display.create {
+        separator = "",
+        items = items,
+        hl_chars = { [os_path_sep] = "TelescopePathSeparator" },
+      }(M.display_items(e))
+    end,
+  }
+end
+
+M.display_items = function(e)
+  local items = M.config.show_scores and { { e.score, "TelescopeFrecencyScores" } } or {}
+
+  -- TODO: store the column lengths here, rather than recalculating in get_display_cols()
+  -- TODO: only include filter_paths column if M.show_filter_col is true
+
+  table.insert(items, {
+    (function()
+      if M.active_filter_tag == "LSP" or M.active_filter_tag == "CWD" then
+        return ts_util.path_tail(M.active_filter) .. os_path_sep
+      elseif M.active_filter then
+        return Path:new(M.active_filter):make_relative(os_home) .. os_path_sep
+      else
+        return ""
+      end
+    end)(),
+    "Directory",
+  })
+
+  table.insert(items, {
+    M.file_format(e.name),
+    util.buf_is_loaded(e.name) and "TelescopeBufferLoaded" or "",
+  })
+
+  if has_devicons and not M.config.disable_devicons then
+    icon, icon_highlight = devicons.get_icon(e.name, string.match(e.name, "%a+$"), { default = true })
+    table.insert(items, { icon, icon_highlight })
+  end
+  return items
+end
+
+---Format filename. Mainly os_home to {~/} or current to {./}
+---@param filename string
+---@return string
+M.file_format = function(filename)
+  local original_filename = filename
+
+  if M.active_filter then
+    filename = Path:new(filename):make_relative(M.active_filter)
+  else
+    filename = Path:new(filename):make_relative(M.cwd)
+    -- check relative to home/current
+    if vim.startswith(filename, os_home) then
+      filename = "~/" .. Path:new(filename):make_relative(os_home)
+    elseif filename ~= original_filename then
+      filename = "./" .. filename
+    end
+  end
+
+  if M.tail_path then
+    filename = util.path_tail(filename)
+  elseif M.shorten_path then
+    filename = util.path_shorten(filename)
+  end
+
+  return filename
+end
```

Comment:
I know this is just moved from old code but it would be cool if this could be integrated into what we are doing with `path_display`. 
See https://github.com/nvim-telescope/telescope.nvim/blob/03c2b8f6bb8027ecdf10acc6d803886c80f99037/doc/telescope.txt#L174-L205
https://github.com/nvim-telescope/telescope.nvim/blob/03c2b8f6bb8027ecdf10acc6d803886c80f99037/doc/telescope_changelog.txt#L50-L86
and the util function

What do you think?

I just think it would be nice if we pickup the configuration the user already defined in telescope.setup

Reply:
connnnni this so amazing 🤩 < tami is so tempted now to go back to work.